wrc/codegen/merger.py

```diff
@@ -0,0 +1,179 @@
+from wrc.sema.ast import Article, Section, TableOfContent, Guideline, split_rule_number
+from copy import deepcopy
+
+NOTES_INDEX = 0
+TOC_INDEX = 1
+FIRST_ARTICLE_INDEX = 2
+
+
+class BadFormatError(Exception):
+    def __init__(self, message: str):
+        self.message = message + '. Please report this issue to the developer.'
+        super().__init__(message)
+
+
+def article_number_is_lower(num1: str, num2: str) -> bool:
+    """
+    Returns True if num1 < num2, otherwise returns False.
+    This is necessary because:
+    * We need to compare article numbers like '1' and 'A'.
+    * '2' > '10' == True, which is a problem.
+    """
```

Comment:
```python3
def article_number_is_lower(num1: str, num2: str) -> bool:
    """
    Returns True if num1 < num2, otherwise returns False.
    This is necessary because:
    * We need to compare article numbers like '1' and 'A'.
    * '2' < '10' is False, which is not what we want.

    We want the following examples to evaluate to True:
    * '5' < '12'.
    * '12' < 'A'.
    * 'A' < 'B'.
    """
    is_digit_1 = num1.isdigit()
    is_digit_2 = num2.isdigit()
    if is_digit_1 and is_digit_2:
        # Both strings contain numbers.
        return int(num1) < int(num2)
    elif is_digit_1 and not is_digit_2:
        # Only num2 is a letter (always number < letter).
        return True
    elif not is_digit_1 and not is_digit_2:
        # Both are letters.
        return num1 < num2

    # The remaining case (num1 letter and num2 number) is always False.
    return False
```